Dashboard/app/js/lib/controllers/survey-controllers.js

```diff
@@ -899,11 +899,260 @@ FLOW.questionControl = Ember.ArrayController.create({
   }.property('content.@each.isSaving')
 });
 
-// TODO turn this into radio buttons
+/*
+ *  Note: This controller is for the option list for a question's dependencies
+ */
 FLOW.optionListControl = Ember.ArrayController.create({
   content: []
 });
 
+/*
+ *  Controller for the list of options attached to an option question
+ *
+ */
+FLOW.questionOptionsControl = Ember.ArrayController.create({
+  content: null,
+  questionId: null,
+
+  /*
+   *  Add a new option object to the content of this controller.  The object
+   *  is not persisted to the data store.
+   */
+  addOption: function() {
+    var c = this.content;
+    c.addObject(Ember.Object.create({
+        code: null,
+        text: null,
+        order: c.get('length') + 1,
+        questionId: this.get('questionId'),
+    }));
+  },
+
+  /*
+   *  Persist all the newly added options to the data store.
+   *  Options with empty code and empty text fields are dropped
+   *  from the list.  If they were already persisted in the datastore
+   *  they are deleted
```

Comment:
The idea is if we have any option in the list with blank text and code, we delete it. But if the blank text and code were from an option that had previously been saved to the datastore, we need to remove it from the datastore as well.  Its a bit of a corner case.